Gems/Terrain/Code/Source/TerrainRenderer/Passes/TerrainClipmapDebugPass.cpp

```diff
@@ -0,0 +1,176 @@
+        if (terrainFeatureProcessor)
+        {
+            const TerrainClipmapManager& clipmapManager = terrainFeatureProcessor->GetClipmapManager();
+            AZ::RHI::FrameGraphAttachmentInterface attachmentDatabase = frameGraph.GetAttachmentDatabase();
+
+            //! If this frame, macro clipmap update is skipped but detail is not,
+            //! then the detail pass will be responsible for importing the clipmaps.
+            if (!clipmapManager.HasMacroClipmapUpdate() && !clipmapManager.HasDetailClipmapUpdate())
+            {
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::MacroColor, attachmentDatabase);
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::MacroNormal, attachmentDatabase);
+            }
+
+            if (!clipmapManager.HasDetailClipmapUpdate())
+            {
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::DetailColor, attachmentDatabase);
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::DetailNormal, attachmentDatabase);
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::DetailHeight, attachmentDatabase);
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::DetailRoughness, attachmentDatabase);
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::DetailSpecularF0, attachmentDatabase);
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::DetailMetalness, attachmentDatabase);
+                clipmapManager.ImportClipmap(TerrainClipmapManager::ClipmapName::DetailOcclusion, attachmentDatabase);
+            }
+
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::MacroColor, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::MacroNormal, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::DetailColor, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::DetailNormal, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::DetailHeight, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::DetailRoughness, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::DetailSpecularF0, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::DetailMetalness, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
+            clipmapManager.UseClipmap(TerrainClipmapManager::ClipmapName::DetailOcclusion, AZ::RHI::ScopeAttachmentAccess::Read, frameGraph);
```

Comment:
It should be possible to give the TerrainClipmapManager more responsibility here so the passes don't have to do this logic. The debug pass shouldn't have to check for what other passes are doing, and this code shouldn't be duplicated between this and the TerrainClipmapCompute pass.

Could there just be a couple calls on the TerrainClipmapManager for `UpdateMacroClipmaps(frameGraph)` and `UpdateDetailClipmaps(frameGraph)` that do this logic? `UpdateDetailClipmaps` could check if the macro clipmaps updated internally to the manager.